### PR TITLE
Add simple Go router based on github.com/julienschmidt/httprouter

### PR DESCRIPTION
**Pull request by radu-matei**
This commit adds a simple Go router, based on https://github.com/julienschmidt/httprouter, which is a zero dependency, high performance router.

This PR also adds type alias for the most commonly used types from the package and a simple example for using the router.

### CODE REVIEW

File: examples/http-tinygo-router/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+$ spin build
+Executing the build command for component tinygo-hello: tinygo build -target=wasi -gc=leaking -no-debug -o main.wasm main.go
+Successfully ran the build command for the Spin components.
+```
+
+Finally, we can create a Spin application configuration to execute this
+component:
+
+```toml
+apiVersion = "0.1.0"
```
Comment (fibonacci1729) [inline examples/http-tinygo-router/README.md:42]:
```suggestion
spin_manifest_version = "1"
```

**Review by fibonacci1729 — APPROVED** [review]

**Comment by radu-matei** [comment]
Example of using the router from this PR: https://github.com/radu-matei/spin-kv-explorer/blob/feat/golang/golang-explorer/main.go

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,7 @@ ignored-assets
 main.wasm
 .parcel-cache
 .vscode/*.log
+go.sum
```
Comment (radu-matei) [inline .gitignore:9]: cc @adamreese — is adding `go.sum` to `.gitignore` the right thing to do here?
I mainly want to make sure we're not polluting the repo with changes here.

File: examples/http-tinygo-router/README.md (hunk 1 above)
Comment (radu-matei) [reply examples/http-tinygo-router/README.md:42]: That's the readme of the example, I'd like to update all of them in one go.

**Pull request merged**